Presentation.Web/app/components/org/user/org-user-delete.controller.ts

```diff
@@ -269,22 +365,52 @@
             };
         }
 
+        private updateGroupSelections() {
+
+            this.checkAndChangeGroupSelectionStatus(this.vmContractRoot);
+            this.checkAndChangeGroupSelectionStatus(this.vmProjectRoot);
+            this.checkAndChangeGroupSelectionStatus(this.vmSystemRoot);
+            this.checkAndChangeGroupSelectionStatus(this.vmDprRoot);
+            this.checkAndChangeGroupSelectionStatus(this.vmOrgRoot);
+
+            const selectedAdminRights = this.collectSelectedAdminRoles();
+            if (selectedAdminRights.length < this.vmAdminRoot.rights.length) {
+                this.vmAdminRoot.selected = false;
+                return;
+            } 
+
+            this.vmAdminRoot.selected = true;
+        }
+
+        private checkAndChangeGroupSelectionStatus(groupRoot: IRootAssignedRightsWithGroupSelectionViewModel) {
+            const selectedContractRights =
+                this.collectSelectedRolesFromSource(groupRoot.rights, (newRights) => this.vmContractRoot.rights = newRights);
```

Comment:
@a-naskret this one is tied to vmContractRoot.. ... Question is what the function call is doing to the rights ... do you know if it makes sense in this contex?